toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Int16.jsm

```diff
@@ -1,4 +1,10 @@
 class {{ ffi_converter }} extends FfiConverter {
+    static checkType(name, value) {
+        super.checkType(name, value);
+        if (!Number.isSafeInteger(value)) {
+            throw TypeError(`${name} exceeds the safe integer bounds (${value})`);
+        }
```

Comment:
Maybe we could just delete the check altogether, since the C++ code does the same thing.  I asked nika that in the phabricator thread, let's follow her advice here.

If we do keep the check, it it seems like this check should be more strict and check if the value will fit in a `i16`.